geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/Geode8119RegressionDistributedTest.java

```diff
@@ -0,0 +1,170 @@
+
+@RunWith(Parameterized.class)
+public class Geode8119RegressionDistributedTest implements Serializable {
+  private static final String REGION_NAME = "TestRegion";
+  private static final String DISK_STORE_ID = "testDisk";
+
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+
+  @Rule
+  public transient GfshCommandRule gfsh = new GfshCommandRule();
+
+  @Rule
+  public DistributedRule distributedRule = new DistributedRule();
+
+  @Rule
+  public SerializableTemporaryFolder temporaryFolder = new SerializableTemporaryFolder();
+
+  @Parameterized.Parameter
+  public RegionShortcut regionShortcut;
+
+  @Parameterized.Parameters(name = "RegionType={0}")
+  public static Iterable<RegionShortcut> data() {
+    return Arrays.asList(RegionShortcut.PARTITION_PERSISTENT, RegionShortcut.REPLICATE_PERSISTENT);
+  }
+
+  private Properties createLocatorConfiguration(int localLocatorPort) {
+    Properties config = new Properties();
+    config.setProperty(MCAST_PORT, "0");
+    config.setProperty(LOCATORS, "localhost[" + localLocatorPort + ']');
+    config.setProperty(START_LOCATOR,
+        "localhost[" + localLocatorPort + "],server=true,peer=true,hostname-for-clients=localhost");
+
+    return config;
+  }
+
+  private Properties createServerConfiguration(int localLocatorPort) {
+    Properties config = new Properties();
+    config.setProperty(MCAST_PORT, "0");
+    config.setProperty(LOCATORS, "localhost[" + localLocatorPort + ']');
+
+    return config;
+  }
+
+  private void createDiskStore(File[] diskStoreDirectories) {
+    DiskStoreFactory diskStoreFactory = cacheRule.getCache().createDiskStoreFactory();
+    diskStoreFactory.setMaxOplogSize(1);
+    diskStoreFactory.setAutoCompact(true);
+    diskStoreFactory.setAllowForceCompaction(true);
+    diskStoreFactory.setDiskDirs(diskStoreDirectories);
+
+    diskStoreFactory.create(DISK_STORE_ID);
+  }
+
+  private void createRegion() {
+    cacheRule.getCache()
+        .<String, String>createRegionFactory(regionShortcut)
+        .setDiskStoreName(DISK_STORE_ID)
+        .create(REGION_NAME);
+  }
+
+  private void createServerWithRegionAndPersistentRegion(File[] diskStoreDirectories) {
+    createDiskStore(diskStoreDirectories);
+    createRegion();
+    cacheRule.getCache().getRegion(REGION_NAME);
+  }
+
+  private void gracefullyDisconnect() {
+    InternalDistributedSystem.getConnectedInstance().stopReconnectingNoDisconnect();
+    InternalDistributedSystem.getConnectedInstance().disconnect();
+    await()
+        .untilAsserted(() -> assertThat(InternalDistributedSystem.getConnectedInstance()).isNull());
+  }
+
+  @Test
+  public void regressionTest() throws IOException {
+    VM locator, server;
+    final int ENTRIES = 100000;
+    final Host host = Host.getHost(0);
+    int site1Port = getRandomAvailableTCPPortsForDUnitSite(1)[0];
+    File diskStoreDirectory1 = temporaryFolder.newFolder("diskDir1");
+    File diskStoreDirectory2 = temporaryFolder.newFolder("diskDir2");
```

Comment:
Use only one dir work. So when you change this part to:
```
    File[] diskStoreDirectories = new File[] {temporaryFolder.newFolder("diskDir1")};

    String diskDirs = Arrays.stream(diskStoreDirectories).map(File::getAbsolutePath)
        .collect(Collectors.joining(","));
```
test is executed successfully.